nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement the correct_name method'
```

Comment:
- **[OPTIONAL]** Specifying the class that has not implemented the `correct_name` method would be nice.

  ```
  raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
  ```

Reply:
@DammyShittu Thank you very mush for this observation. I am obviously adding this as guided.